src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -1,74 +1,86 @@
 
 public class Shooter extends SubsystemBase {
-    
-    private static Shooter shooter; 
-     public static Shooter getInstance()
-    {
-        if (shooter == null)
-            shooter = new Shooter();
- 
-        return shooter;
-    }
-
     private CANSparkMax m_motor;
     private SparkPIDController m_pidController;
     private RelativeEncoder m_encoder;
 
-    private double kP, kD, kFF, kMaxOutput, kMinOutput, maxRPM; 
-    
-        // add Tunable Number later for ajusting Rpm on the flywheels 
-        public Shooter() {  
-            m_motor = new CANSparkMax(32, MotorType.kBrushless);
-
-            m_motor.restoreFactoryDefaults();
-            m_motor.setSmartCurrentLimit(70);
-            m_pidController = m_motor.getPIDController();
+    private TunableNumber kP = new TunableNumber("Shooter/kP", Config.ShooterConstants.kP);
+    private TunableNumber kI = new TunableNumber("Shooter/kI", Config.ShooterConstants.kI);
+    private TunableNumber kD = new TunableNumber("Shooter/kD", Config.ShooterConstants.kD);
+    private TunableNumber kFF = new TunableNumber("Shooter/kFF", Config.ShooterConstants.kFF);
+ 
+    private static Shooter shooter;
+    public static Shooter getInstance() {
+        if (shooter == null)
+            shooter = new Shooter();
+        return shooter;
+    }
 
-            m_encoder = m_motor.getEncoder();
+    // add Tunable Number later for ajusting Rpm on the flywheels
+    public Shooter() {
+        System.out.println("[Init] Creating Shooter");
+        m_motor = new CANSparkMax(Config.ShooterConstants.FLY_WHEEL_ID, MotorType.kBrushless);
+        m_motor.restoreFactoryDefaults();
 
-            //confi
-            kP = 6e-5;
-            kD = 0;
-            kFF = 0.000015;
-            kMaxOutput = 1;
-            kMinOutput = -1;
-            maxRPM = 5700; 
+        m_motor.setCANTimeout(500); //Units in miliseconds
+        m_motor.setIdleMode(IdleMode.kCoast);
+        m_motor.setInverted(false);
 
-            m_pidController.setP(kP);
-            m_pidController.setD(kD);
-            m_pidController.setFF(kFF);
-            m_pidController.setOutputRange(kMaxOutput, kMinOutput);
-        }
+        m_pidController = m_motor.getPIDController();
+        m_encoder = m_motor.getEncoder();
+        m_encoder.setAverageDepth(2);//check if 1 would work better
 
-        public double getVelocity() {
+        //Voltage compensation
+        m_motor.enableVoltageCompensation(12); 
+        m_motor.setSmartCurrentLimit(40);  
+        setBrake(false);
 
-                return m_encoder.getVelocity();
+        m_pidController.setOutputRange(Config.ShooterConstants.kMinOutput, Config.ShooterConstants.kMaxOutput);
+        setPIDGains(kP.get(), kI.get(), kD.get());
+        setFFGains(kFF.get());
+    }
 
-        }
+    public double getVelocity() {
+        return m_encoder.getVelocity();
+    }
 
-        public void setSetPoint(double setPoint){
+    public void setSetPoint(double setPoint) {
```

Comment:
It is better to change its name to setSetRPM or something related to velocity/RPM